File: src/main/java/de/grimsi/gameyfin/layouts/SetupLayout.java
package de.grimsi.gameyfin.layouts;

import com.flowingcode.vaadin.addons.fontawesome.FontAwesome;
import com.vaadin.flow.component.applayout.AppLayout;
import com.vaadin.flow.component.button.Button;
import com.vaadin.flow.component.button.ButtonVariant;
import com.vaadin.flow.component.dependency.CssImport;
import com.vaadin.flow.component.html.Image;
import com.vaadin.flow.component.orderedlayout.FlexComponent;
import com.vaadin.flow.component.orderedlayout.HorizontalLayout;
import de.grimsi.gameyfin.resources.PublicResources;
import de.grimsi.gameyfin.services.ThemeService;
import org.springframework.beans.factory.annotation.Autowired;

@CssImport("./styles/header.css")
public class SetupLayout extends AppLayout {

    public SetupLayout(@Autowired ThemeService themeService) {
        Image logo = new Image(PublicResources.GAMEYFIN_LOGO.path, "Gameyfin Logo");
        logo.addClassName("header-logo");

        Button toggleTheme = new Button(FontAwesome.Solid.CIRCLE_HALF_STROKE.create());
        toggleTheme.addThemeVariants(ButtonVariant.LUMO_ICON);
        toggleTheme.addClickListener(listener -> themeService.toggleTheme());

        HorizontalLayout horizontalLayout = new HorizontalLayout();
        horizontalLayout.setWidthFull();
        horizontalLayout.setAlignSelf(FlexComponent.Alignment.END);
        horizontalLayout.add(logo, toggleTheme);

        addToNavbar(horizontalLayout);
    }
}

File: src/main/kotlin/de/grimsi/gameyfin/ui/layouts/SetupLayout.kt
package de.grimsi.gameyfin.ui.layouts

import com.flowingcode.vaadin.addons.fontawesome.FontAwesome
import com.github.mvysny.karibudsl.v10.*
import com.vaadin.flow.component.UI
import com.vaadin.flow.component.orderedlayout.FlexComponent
import com.vaadin.flow.router.RouterLayout
import de.grimsi.gameyfin.setup.SetupService
import de.grimsi.gameyfin.ui.resources.PublicResources
import de.grimsi.gameyfin.ui.services.ThemeService
import de.grimsi.gameyfin.ui.views.LoginView
import org.springframework.beans.factory.annotation.Autowired

class SetupLayout(
    @Autowired private val setupService: SetupService,
    @Autowired private val themeService: ThemeService
) : KComposite(), RouterLayout {

    init {
        if (setupService.isSetupCompleted()) UI.getCurrent().navigate(LoginView::class.java)

        ui {
            appLayout {
                navbar {
                    flexLayout {
                        setWidthFull()
                        alignItems = FlexComponent.Alignment.CENTER

                        image(PublicResources.GAMEYFIN_LOGO.path) {
                            setWidthFull()
                            height = "40px"
                            className = "header-logo"
                        }

                        horizontalLayout {
                            alignItems = FlexComponent.Alignment.CENTER

                            val toggleDarkModeIcon =
                                FontAwesome.Solid.CIRCLE_HALF_STROKE.create { _ -> themeService.toggleTheme() }
                            iconButton(toggleDarkModeIcon)
                        }
                    }
                }
            }
        }
    }
}